src/utils/polyfillEnvironment.js

```diff
@@ -36,21 +36,31 @@ require('InitializeCore'); // eslint-disable-line import/no-unresolved
 
 require('../hot/client/importScriptsPolyfill');
 
-const scriptURL = require('react-native').NativeModules.SourceCode.scriptURL; // eslint-disable-line import/no-unresolved
-// In order to ensure hot client has a valid URL we need to get a valid origin
-// from URL from which the bundle was loaded. When using iOS simulator/Android emulator
-// or Android device it will be `localhost:<port>` but when using real iOS device
-// it will be `<ip>.xip.io:<port>`. Thus the code below ensure we connect and download
-// manifest/hot-update from a valid origin.
-let devServerOrigin = null;
-
-if (scriptURL) {
-  const [protocol, , origin] = scriptURL.split('/');
-  devServerOrigin = `${protocol}//${origin}`;
-}
+if (process.env.NODE_ENV !== 'production') {
+  let protocol;
+  let origin;
+
+  // If remote debugger is attached, we have access to `window` object
+  // from which we can get `protocol` and `origin` of dev server.
+  if (typeof window !== 'undefined' && window.location) {
```

Comment:
Can we do this as the fallback (do we actually need to?) instead of the preferred method?

Reply:
No it must be a preferred way otherwise, this fallback will never gets executed.

Reply:
@zamotany can `scriptURL` return wrong value when you're debugging? can you add a comment on why this is the case? because I have no idea.

Reply:
`scriptURL`: `http://10.0.2.2:8081/...` vs debugger host `http://localhost:8081`. You need to use `localhost` because `importScripts` is executed in debugger context. Otherwise you would get CSP error.

Reply:
@zamotany thanks! add a comment please :D